diff --git a/Data/NosGm.GameObject/Mate.cs b/Data/NosGm.GameObject/Mate.cs
index fa2c76a..0000000 100644
--- a/Data/NosGm.GameObject/Mate.cs
+++ b/Data/NosGm.GameObject/Mate.cs
@@ -932,13 +932,14 @@ namespace NosGm.GameObject
             else if (skill != null && skill.TargetType == 0 && session.HasCurrentMapInstance)
             {
                 MapMonster monsterToAttack = attacker.Owner.MapInstance.GetMonsterById(targetId);
-                if (monsterToAttack == null || attacker.Mp <= skill.MpCost)
+                if (monsterToAttack == null || attacker.Mp < skill.MpCost)
                 {
                     return;
                 }
 
-                if (Map.GetDistance(new MapCell { X = monsterToAttack.MapX, Y = monsterToAttack.MapY }, new MapCell { X = monsterToAttack.MapX, Y = monsterToAttack.MapY }) >=
+                if (Map.GetDistance(new MapCell { X = attacker.PositionX, Y = attacker.PositionY },
+                        new MapCell { X = monsterToAttack.MapX, Y = monsterToAttack.MapY }) >=
                     skill.Range + 1 + monsterToAttack.Monster.BasicArea)
                 {
                     return;
@@ -1113,11 +1114,11 @@ namespace NosGm.GameObject
                         $"{Level} " +
                         $"{Loyalty} " +
                         $"{Experience} " +
+                        $"{Monster.AttackClass} " +
                         $"{Attack} " +
-                        $"{Defence} " +
                         $"{DamageMinimum} " +
                         $"{DamageMaximum} " +
                         $"{Concentrate} " +
                         $"{Monster.CriticalChance} " +
                         $"{Monster.CriticalRate} " +
-                        "0 " +
+                        $"{Defence} " +
                         $"{MeleeDefense} " +
                         $"{MeleeDefenseDodge} " +
                         $"{RangeDefense} " +
@@ -1132,18 +1133,11 @@ namespace NosGm.GameObject
                         $"{MaxHp} " +
                         $"{Mp} " +
                         $"{MaxMp} " +
-                        "1 " +
+                        $"{(IsTeamMember ? 1 : 0)} " +
                         $"{xp} " +
                         $"{(CanPickUp ? 1 : 0)} " +
                         $"{Name.Replace(' ', '^')} " +
                         $"{(IsSummonable ? 1 : 0)} " +
-                        $"0 " +
-                        $"{StarRating} " +
-                        $"{TrainerLevel} " +
-                        $"{Skill1} " +
-                        $"{Skill2} " +
-                        $"{CurrentEXP} " +
-                        $"{(EXPNeeded == 0 ? 1 : EXPNeeded)} ";
+                        "0";
             }
 
             return "";
